sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -0,0 +1,751 @@
+        `None` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        `-1` means using all processors. See :term:`Glossary <n_jobs>`
+        for more details.
+
+    cluster_selection_method : {"eom", "leaf"}, default="eom"
+        The method used to select clusters from the condensed tree. The
+        standard approach for HDBSCAN* is to use an Excess of Mass (`"eom"`)
+        algorithm to find the most persistent clusters. Alternatively you can
+        instead select the clusters at the leaves of the tree -- this provides
+        the most fine grained and homogeneous clusters.
+
+    allow_single_cluster : bool, default=False
+        By default HDBSCAN* will not produce a single cluster, setting this
+        to True will override this and allow single cluster results in
+        the case that you feel this is a valid result for your dataset.
+
+    store_centers : str, default=None
+        Which, if any, cluster centers to compute and store. The options are:
+        - `None` which does not compute nor store any centers.
+        - `"centroid"` which calculates the center by taking the weighted
+          average of their positions. Note that the algorithm uses the
+          euclidean metric and does not guarantee that the output will be
+          an observed data point.
+        - `"medoid"` which calculates the center by taking the point in the
+          fitted data which minimizes the distance to all other points in
+          the cluster. This is slower than "centroid" since it requires
+          computing additional pairwise distances between points of the
+          same cluster but guarantees the output is an observed data point.
+          The medoid is also well-defined for arbitrary metrics, and does not
+          depend on a euclidean metric.
+        - `"both"`which computes and stores both forms of centers.
+
+    copy : bool, default=False
+        If `copy=True` then any time an in-place modifications would be made
+        that would overwrite data passed to :term:`fit`, a copy will first be
+        made, guaranteeing that the original data will be unchanged. Currently
+        this only makes a difference when passing in a dense precomputed
+        distance array (i.e. when `metric="precomputed"`) and using the
+        `"brute"` algorithm (see `algorithm` for details).
+
+    Attributes
+    ----------
+    labels_ : ndarray of shape (n_samples,)
+        Cluster labels for each point in the dataset given to :term:`fit`.
+        There are two reserved labels:
+        - Noisy samples are given the label -1.
+        - Samples with missing data are given the label -2.
+
+    probabilities_ : ndarray of shape (n_samples,)
+        The strength with which each sample is a member of its assigned
+        cluster. Noise points have probability zero; points in clusters
+        have values assigned proportional to the degree that they
+        persist as part of the cluster.
```

Comment:
I think samples with missing values will have nan probabilities. May we document this?